src/Linters/BaseLinter.hack

```diff
@@ -9,11 +9,16 @@
 
 namespace Facebook\HHAST;
 
+use namespace Facebook\TypeAssert;
 use type Facebook\HHAST\File;
 use namespace HH\Lib\{C, Str};
 
 <<__ConsistentConstruct>>
 abstract class BaseLinter {
+  <<__Reifiable>>
+  const type TConfig as mixed = mixed;
```

Comment:
Maybe use `shape(...)` as the base type? Or are there any cases where we wouldn't want the config to be a shape?